Mathlib/LinearAlgebra/BilinearMap.lean

```diff
@@ -357,6 +393,31 @@ def compr₂ (f : M →ₗ[R] Nₗ →ₗ[R] Pₗ) (g : Pₗ →ₗ[R] Qₗ) : M
 theorem compr₂_apply (f : M →ₗ[R] Nₗ →ₗ[R] Pₗ) (g : Pₗ →ₗ[R] Qₗ) (m : M) (n : Nₗ) :
     f.compr₂ g m n = g (f m n) := rfl
 
+/-- A version of `Function.Injective.comp` for composition of a bilinear map with a linear map. -/
+theorem injective_compr₂_of_injective (f : M →ₗ[R] Nₗ →ₗ[R] Pₗ) (g : Pₗ →ₗ[R] Qₗ) (hf : Injective f)
+    (hg : Injective g) : Injective (f.compr₂ g) := by
```

Comment:
Argubly we should state this in the reduced form
```suggestion
    (hg : Injective g) : Injective (fun x => g.comp (f x)) := by
```
which can also then be written for semilinear maps.